servicemanager/servicemanager.go

```diff
@@ -285,7 +295,32 @@ func (sm *ServiceManager) ValidateService(service ServiceInfo) error {
  * Private
  **********************************************************************************************************************/
 
+func (sm *ServiceManager) doRestoreService(serviceID string) error {
+	log.WithFields(log.Fields{"serviceID": serviceID}).Debug("Restore service")
+
+	service, err := sm.serviceInfoProvider.GetService(serviceID)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	if !service.Cached {
+		log.Warningf("Service %v not cached", serviceID)
+
+		return nil
+	}
+
+	sm.serviceAllocator.RestoreOutdatedItem(serviceID)
```

Comment:
I suggest to combine serviceAllocator.RestoreOutdatedItem and sm.serviceInfoProvider.SetServiceCached into one function.